### PR TITLE
[DO NOT SQUASH MERGE] Mdn folder to markdown conversion

### LINKED ISSUE #9106: [Markdown] Prepare /mdn docs for markdowning
This is a tracking bug for the work we have to do to prepare the /mdn docs for Markdown conversion.

* [ ] remove `id` attributes except on headings:
* [ ] deal with hidden code blocks in live samples
* [ ] remove hidden elements that don't contain code blocks in a live sample: https://github.com/mdn/content/pull/8962
* [ ] remove summary and seoSummary class: https://github.com/mdn/content/pull/8964
* [ ] remove inline styles: https://github.com/mdn/content/pull/8966
* [ ] make note/warning consistently formatted
* [ ] fix broken dl elements: https://github.com/mdn/content/pull/8963
* [ ] deal with sup and sub
* [ ] fix live samples that use IDs which are not simple derivations of heading text
* [ ] remove random CSS classes
* [ ] remove unsupported elements
* [ ] ensure tables are converted appropriately
* [ ] fix broken `<strong><code>` markup, and other inline markup issues

### PR DESCRIPTION
**Pull request by Rumyra**
#### Summary

The /mdn/ folder converted to markdown

#### Related issues

Fixes #9106

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

I reviewed all the pages and they all render ok 👍

### CODE REVIEW

**Comment by wbamberg** [comment]
In this PR I only see 1 commit for the conversion (https://github.com/mdn/content/pull/9589/commits/efc1210e4ee2446ced3f7df96313f505783e4ecd).

The conversion needs to be in 2 separate commits:
* commit 1 is to rename all the files from HTML->MD
* commit 2 does the conversion of the file contents

I'm reliably informed that this is the trick that enables us to preserve Git history across the files.

The converter h2m makes this easy for you by doing the rename, then staging, then doing the conversion. So you can do something like:

(from yari)
```
yarn md h2m mdn --mode==replace
```

...then (from content)
```
git commit -S -m "Rename HTML files to MD"
git add files/en-us/mdn
git commit -S -m "Convert file contents to Markdown"
git log --oneline             <- should show you 2 commits
```

Does that make sense?

**Comment by hamishwillee** [comment]
@Rumyra This appears to have the same problem as @wbamberg had a few builds ago - the bullets are all indented 2 spaces. I think he fixed using a more recent version of yari, but it might have been a separate run of prettify.

File: files/en-us/mdn/structures/page_types/api_reference_page_template/index.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
```
Comment (hamishwillee) [inline files/en-us/mdn/structures/page_types/api_reference_page_template/index.md:1]: It's a pity I didn't get through these templates - most of them I did manually and they are much better conversions than this - e.g.  https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_property_subpage_template

**Comment by wbamberg** [comment]
> @Rumyra This appears to have the same problem as @wbamberg had a few builds ago - the bullets are all indented 2 spaces. I think he fixed using a more recent version of yari, but it might have been a separate run of prettify.

Yes, I just updated my Yari and tried again.

**Pull request closed without merging**